src/interface/container.hpp

```diff
@@ -71,7 +75,10 @@ class Container {
 
   ///
   /// Set the pointer to the mesh block for this container
-  void setBlock(MeshBlock *pmb) { pmy_block = pmb; }
+  void setBlock(std::weak_ptr<MeshBlock> pmb) { pmy_block = pmb; }
+  void setBlock(const std::shared_ptr<Container<T>> &other) {
```

Comment:
[nitpicking]
Should be potentially call this SetBlockPointer (so that it matches GetBlockPointer).
As far as I see this is only called like 5 times in the codebase so the extra characters wouldn't clutter too much
[/nitpicking]

Reply:
Done.